### PR TITLE
Dependencies

### PR DESCRIPTION
**Pull request by ariedov**
updates dependencies and fixes code to make it build and work with Android Studio 3.0

### CODE REVIEW

**Comment by ariedov** [comment]
i can squash it to a branch and create new pull request if necessary

File: app/src/main/java/com/derek_s/hubble_gallery/base/Constants.java [hunk 1]
```diff
@@ -12,8 +14,11 @@
   public static final int FAVORITES_MODE = 1;
   public static final int SEARCH_MODE = 2;
 
-  public static String imageDirectory() {
+  public static String externalImageDirectory() {
     return Environment.getExternalStorageDirectory() + "/Pictures/Hubble";
   }
 
+  public static String internalImageDirectory(File rootDir) {
```
Comment (derekcsm) [inline app/src/main/java/com/derek_s/hubble_gallery/base/Constants.java:21]: smart 👍

File: app/src/main/kotlin/com/derek_s/hubble_gallery/detailspage/DetailsPresenter.java [hunk 2]
```diff
@@ -31,25 +31,26 @@
...
-  public int lightVibrantColor;
-  public int toolbarBgColorAlpha;
-  @Inject
-  Context context;
-  @Inject
-  Resources resources;
-  @Inject
-  FavoriteUtils favoriteUtils;
-  private String TAG = getClass().getSimpleName();
+  private static final String TAG = DetailsPresenter.class.getSimpleName();
```
Comment (derekcsm) [inline app/src/main/kotlin/com/derek_s/hubble_gallery/detailspage/DetailsPresenter.java:34]: oh man you're a saint, this was so messy before

**Comment by derekcsm** [comment]
Just looked over, really love this man, some of the refactoring decisions we're exactly how I wanted them done but was just too lazy to do myself xD - thank you so much! 

I'll merge to dev, then test/update some things - my goal is to release an update end of week because of this now.

**Comment by ariedov** [comment]
thank you! I am glad you liked it.

i am also planning to get rid of the collapsable recycle view and toasty.
all of it can be replaced with stuff from the design library

**Comment by derekcsm** [comment]
Yeah Toasty is unnecessary I don't use that pattern anymore, be warned the collapsing recycler is a tad complicated. If you're interested I think one really annoying UI thing in the app is the collapsing Toolbar in the main view that isn't based on how those are supposed to work (it moves the view instead of just itself so all items in the Recycler get pushed around when it opens/closes), I'd like to use a better toolbar design there.

**Comment by ariedov** [comment]
The toolbar can be fixed using CoordinatorLayout and AppBarLayout from the design library. I'll try to take care of it too.

File: app/src/main/kotlin/com/derek_s/hubble_gallery/detailspage/DetailsPresenter.java (hunk 2 above)
Comment (ariedov) [reply app/src/main/kotlin/com/derek_s/hubble_gallery/detailspage/DetailsPresenter.java:34]: Ha-ha! Saint =) I'll print it out and put in a frame =)

**Pull request merged**